lib/renderers/DefaultRenderer.js

```diff
@@ -68,38 +67,72 @@ class DefaultRenderer {
     return this.renderMessage(msg, alexaEvent);
   }
 
-  renderMessage(msg, alexaEvent) {
-    const toRender = ['ask', 'tell', 'say', 'reprompt', 'card.title', 'card.content', 'card.text', 'card.image.smallImageUrl', 'card.image.largeImageUrl'];
-
-    return Promise.all(_(toRender)
-      .map((key) => {
-        let statement = _.at(msg, key)[0];
-        if (!statement) return null;
-        if (_.isArray(statement)) statement = _.sample(statement);
-        return this.renderStatement(statement, alexaEvent)
-          .then(rendered => _.set(msg, key, rendered));
-      })
-      .compact()
-      .value())
-      .then(() => msg);
-  }
+  renderMessage(msg, event) {
+    function deepSearchStatement(statement, alexaEvent) {
```

Comment:
Let's got some clearer comments and start using better named functions to explain what we're doing here. Also unit tests. Also, how much validation should we do on the render cmd object? Before we only rendered certain fields, but I wonder now if we just render everything.

Reply:
I create new function with better names